.github/scripts/build.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -ex
+
+environment=$1
+
+cd notary-server
+cargo build --release
+aws s3 cp target/release/notary-server s3://tlsn-deploy/$environment/
```

Comment:
How is this bucket  managed?

We could also use https://github.com/actions/upload-artifact
which offers the ` retention-days:` option.

Reply:
This resource was created and it' s managed by Terraform.
There is no lifecycle policy at the moment, but it wouldn' t be a problem to create one.

Reply:
Can we add a comment here very briefly explaining why we are doing this S3 binary copy step?

Reply:
We use an S3 bucket because it' s the most straightforward and effective method for our case. Another approach could be uploading artifacts to GH using that specific action mentioned above, which might be convenient if you want to directly access the binary from this particular workflow. However, if you retrieve it from a 3rd party instance, it becomes more complicated. In our case, the workflows are executed on GH runners and have the necessary permissions to trigger deployment against our hosted AWS instance. To achieve this using GH artifacts, we would need to use the GH API and a token (running on the EC2 instance), which can be somehow complicated.
We can give it a try if you think it makes sense (the last time we attempted this was a year ago).

Reply:
thanks for the detailed explanation! aha I actually meant to put a small link to this PR on the script

Reply:
https://github.com/ntampakas/tlsn/commit/48a9f8e1b421b92d5145a4904b808d10a237617a